### PR TITLE
add foregroundServiceType to services using Bluetooth

### LINKED ISSUE #1566: xDrip application stops receiving data
### Subject of the issue
xDrip foreground services seems to stop, no data is received in xDrip.

### Your environment
**xDrip version:** 2020.12.18
**Datasource:** Libre2 (patched app)
**Device:** Xiaomi Mi9

### Expected behavior
xDrip app should keep running in background, receive and show data sent by the Libre2 (patched) app.

### Actual behavior
In Events log shows this:

**FOREGROUND**
```
Got exception trying to use Android 10+ service starting for DoNothingService java.lang.IllegalArgumentException:
foregroundServiceType 0x00000008 is not a subset of foregroundServiceType attribute 0x00000000 in service element of manifest file
```

### PR DESCRIPTION
**Pull request by tolot27**
With Android 10+ some privacy features have been [implemented](https://developer.android.com/about/versions/10/privacy/changes), especially, background services accessing Bluetooth devices also need [location permissions](https://developer.android.com/about/versions/10/privacy/changes#app-access-device-location).

If this permission is not granted, exceptions are thrown as reported with #1566 and as I discovered with InPen authorization. Because we are using compileSdkVersion 29 (even if our targetSdkVersion is 23) we have to the foregroundServiceType.

This PR adds this to all services using Bluetooth and this should fix #1566.

### CODE REVIEW

**Review by alxsimo — APPROVED** [review]

**Comment by tolot27** [comment]
@alxsimo Please do not approve PRs unless you have reviewed it carefully. Is is also highly recommend to leave comments on approval.

**Comment by alxsimo** [comment]
@tolot27 changes are minor in this PR, reviewed them, it's just XML.

Also checked if there could be needed any other params from the [official docs](https://developer.android.com/guide/topics/manifest/service-element).

Any way, only count reviews from people with write access, didn't knew that, sorry

**Comment by tim162** [comment]
@tolot27 @alxsimo I tested the changes in my branch and it seems to resolve the issue. I've used it for 2 hours and didn't get the ForegroundService error in my logs and no data loss. I'll keep you updated if it occurs anytime.

**Review by jamorham — APPROVED** [review]
I tried this out on some of the other services recently and it doesn't seem to do any harm and may be needed on certain handsets. I did also find that it doesn't seem to be consistently implemented in the framework but that doesn't cause any harm. I will merge for wider testing.

**Pull request merged**
